cmd/release/cmd/tag.go

```diff
@@ -204,13 +208,79 @@ var systemAgentInstallerK3sTagSubCmd = &cobra.Command{
 	},
 }
 
+var uiTagSubCmd = &cobra.Command{
+	Use:   "ui [ga,rc] [version]",
+	Short: "Tag ui releases",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if len(args) < 2 {
+			return errors.New("expected at least two arguments: [ga,rc] [version]")
+		}
+
+		rc, err := releaseTypePreRelease(args[0])
+		if err != nil {
+			return err
+		}
+
+		tag := args[1]
```

Comment:
Let's add some validation here like:

```go
if tag != "ga" && tag != "rc" { return errors.New("invalid tag given") }
```

Reply:
This tag is actually a semver, a few lines above it we have:
```golang
rc, err := releaseTypePreRelease(args[0])
if err != nil {
	return err
}
```

And `releaseTypePreRelease(...)` checks if it's a `ga`, `rc`, `alpha`, `debug`, etc.

Reply:
@rafaelbreno 
What do you think about moving the validation functino `releaseTypePreRelease` to the standard
`ValidArgsFunction` from cobra, like this:
https://github.com/rancher/ecm-distro-tools/blob/master/cmd/release/cmd/push.go#L47

Reply:
We should validate the input before the data gets passed to the function that needs it. Ie validate as soon as it's received and error out immediately.